src/sentry/integrations/opsgenie/utils.py

```diff
@@ -20,12 +29,31 @@
 def build_incident_attachment(
     incident: Incident,
     new_status: IncidentStatus,
-    metric_value: float,
+    metric_value: float | None = None,
     notification_uuid: str | None = None,
 ) -> dict[str, Any]:
+    if metric_value is None:
+        metric_value = get_metric_count_from_incident(incident)
+
     data = incident_attachment_info(
-        incident, new_status, metric_value, notification_uuid, referrer="metric_alert_opsgenie"
+        AlertContext(
+            name=incident.alert_rule.name,
+            action_identifier_id=incident.alert_rule.id,
+            threshold_type=AlertRuleThresholdType(incident.alert_rule.threshold_type),
+            detection_type=AlertRuleDetectionType(incident.alert_rule.detection_type),
+            comparison_delta=incident.alert_rule.comparison_delta,
+        ),
```

Comment:
Could we add a convenience method on the `AlertContext` class to do this construction from an incident? It'll cut down the chance that a typo sneaks in.